src/core/targeting/build-page-targeting.ts

```diff
@@ -70,13 +71,35 @@ const filterValues = (pageTargets: Record<string, unknown>) => {
 	return filtered;
 };
 
+const getLastNavigationType = (): NavigationTimingType | undefined => {
+	if (!supportsPerformanceAPI()) {
+		return undefined;
+	}
+	const navigationEvents = performance.getEntriesByType('navigation');
+	const lastNavigationEvent = navigationEvents[navigationEvents.length - 1];
+	return lastNavigationEvent?.type;
+};
+
+const referrerMatchesHost = (referrer: string): boolean => {
+	if (!referrer) {
+		return false;
+	}
+	const referrerUrl = new URL(referrer);
+	return referrerUrl.hostname === window.location.hostname;
+};
+
 // A consentless friendly way of determining if this is the users first visit to the page
 const isFirstVisit = (referrer: string): boolean => {
```

Comment:
I don't know if you think the following is better or worse, but I think it's functionally the same. Will leave it up to you.

const isFirstVisit = (referrer: string): boolean => {
if (supportsPerformanceAPI() && getLastNavigationType() !== 'navigate' ) {
		return false;
	}

return !referrerMatchesHost(referrer)
}